lib/fluent/plugin/out_gcs.rb

```diff
@@ -126,19 +148,29 @@ def hex_random(chunk)
       Digest::MD5.hexdigest(chunk.unique_id)[0...@hex_random_length]
     end
 
-    def format_path(chunk)
-      now = Time.strptime(chunk.key, @time_slice_format)
+    def format_path(time_slice_format, time_slice)
+      time_slice_format = @configured_time_slice_format || timekey_to_timeformat(@buffer_config['timekey'])
+
+      now = Time.strptime(time_slice, time_slice_format)
       (@localtime ? now : now.utc).strftime(@path)
     end
 
     def generate_path(chunk, i = 0, prev = nil)
+      metadata = chunk.metadata
+      time_slice_format = @configured_time_slice_format || timekey_to_timeformat(@buffer_config['timekey'])
+
+      time_slice = if metadata.timekey.nil?
+                     ''.freeze
+                   else
+                     Time.at(metadata.timekey).utc.strftime(time_slice_format)
+                   end
       tags = {
         "%{file_extension}" => @object_creator.file_extension,
         "%{hex_random}" => hex_random(chunk),
         "%{hostname}" => Socket.gethostname,
         "%{index}" => i,
-        "%{path}" => format_path(chunk),
-        "%{time_slice}" => chunk.key,
+        "%{path}" => format_path(time_slice_format, time_slice),
```

Comment:
Hmm, you are right.
We can use `extract_placeholders` here like as fluent-plugin-s3.